File: core/SyncPlatform/src/Services/Exercise/Exercise.Domain/Models/WorkoutTemplate.cs
using MongoDB.Bson.Serialization.Attributes;
using Libs.Shared.Enums;

namespace Exercise.Domain.Models;

public class WorkoutTemplate : BaseMongoEntity
{
    public string Name { get; set; } = string.Empty;

    public string Goal { get; set; } = string.Empty;

    public Difficulty Difficulty { get; set; }

    public int EstimatedDurationMinutes { get; set; }

    public List<string> TargetMuscleGroups { get; set; } = [];

    public List<string> RequiredEquipment { get; set; } = [];

    public int EstimatedCaloriesBurn { get; set; }

    public int AiRecoveryScore { get; set; }

    public bool IsSystemTemplate { get; set; }

    public string CreatedBy { get; set; } = string.Empty;

    public List<TemplateSessionBlock> Sessions { get; set; } = [];

    public class TemplateSessionBlock
    {
        public int Order { get; set; }

        public Guid ExerciseId { get; set; }

        public int Sets { get; set; }

        public int MinReps { get; set; }

        public int MaxReps { get; set; }

        public int RestSeconds { get; set; }

        public string Tempo { get; set; } = string.Empty;

        public int Rir { get; set; }

        [BsonIgnoreIfNull]
        public string? Notes { get; set; }
    }
}

File: ui/web/src/lib/validations/workout-template.ts
import { z } from "zod";

export const sessionBlockSchema = z.object({
  order:       z.coerce.number().int().min(0),
  exerciseId:  z.string().min(1, "Required"),
  sets:        z.coerce.number().int().min(0),
  minReps:     z.coerce.number().int().min(0),
  maxReps:     z.coerce.number().int().min(0),
  restSeconds: z.coerce.number().int().min(0),
  tempo:       z.string().default(""),
  rir:         z.coerce.number().int().min(0),
  notes:       z.string().optional(),
});

export const workoutTemplateSchema = z.object({
  name:                     z.string().min(1, "Required").max(256),
  goal:                     z.string().min(1, "Required").max(256),
  difficulty:               z.string().min(1, "Required"),
  estimatedDurationMinutes: z.coerce.number().int().min(0),
  estimatedCaloriesBurn:    z.coerce.number().int().min(0),
  aiRecoveryScore:          z.coerce.number().int().min(0).max(100),
  isSystemTemplate:         z.boolean().default(true),
  createdBy:                z.string().min(1).default("system"),
  sessions:                 z.array(sessionBlockSchema).default([]),
});

export type WorkoutTemplateFormValues = z.infer<typeof workoutTemplateSchema>;
